### PR TITLE
Modify safeupdate usage to fit with extension changes

### LINKED ISSUE #1308: Cannot use safeupdate for the database nor the postgres user (session_preload_libraries)
# Bug report

Basically, this worked in older projects. In newer projects, the permissions are stricter and one cannot enable this, safeguarding, `safeupdate` extension. But this is a crucial help for avoiding human failure.

```sql
-- 1. doesn't work
GRANT SET ON PARAMETER session_preload_libraries TO postgres;
ALTER ROLE postgres SET session_preload_libraries = 'safeupdate';

-- 2. also doesn't work
alter database postgres set session_preload_libraries = 'safeupdate';


-- test
CREATE TABLE IF NOT EXISTS todos (id SERIAL NOT NULL PRIMARY KEY, title TEXT);
INSERT INTO todos (title) VALUES ('test-todo-123');

DELETE FROM todos; -- should fail with safeupdate
```

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

It's all said.

## To Reproduce

Try either of the provided options in the intro.

## Expected behavior

`safeupdate` can be enabled in the database (I don't think we necessarily need `postgres` role setting but rather one config that enables this in the database or not).

## Screenshots

image

### LINKED ISSUE #27455: Cannot use safeupdate for the database nor the postgres user (session_preload_libraries)
# Bug report

Basically, this worked in older projects. In newer projects, the permissions are stricter and one cannot enable this, safeguarding, `safeupdate` extension. But this is a crucial help for avoiding human failure.

```sql
-- 1. doesn't work
GRANT SET ON PARAMETER session_preload_libraries TO postgres;
ALTER ROLE postgres SET session_preload_libraries = 'safeupdate';

-- 2. also doesn't work
alter database postgres set session_preload_libraries = 'safeupdate';


-- test
CREATE TABLE IF NOT EXISTS todos (id SERIAL NOT NULL PRIMARY KEY, title TEXT);
INSERT INTO todos (title) VALUES ('test-todo-123');

DELETE FROM todos; -- should fail with safeupdate
```

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

It's all said.

## To Reproduce

Try either of the provided options in the intro.

## Expected behavior

`safeupdate` can be enabled in the database (I don't think we necessarily need `postgres` role setting but rather one config that enables this in the database or not).

## Screenshots

image

### PR DESCRIPTION
**Pull request by encima**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

docs update

## What is the current behavior?

Current behaviour is inconsistent across project versions and can result in permissions errors

## What is the new behavior?

New behaviour loads `safeupdate` by default into `shared_preload_libraries` (for all versions after 15.1.173) and can then be enabled on a per-role basis.

Closes supabase/postgres#1308

### CODE REVIEW

File: apps/docs/content/guides/database/extensions/pg-safeupdate.mdx [hunk 1]
```diff
@@ -10,19 +10,17 @@ The `pg-safeupdate` extension is a useful tool for protecting data integrity and
...
-
-{/* prettier-ignore */}
-```sql
-GRANT SET ON PARAMETER session_preload_libraries TO anon;
-ALTER ROLE anon SET session_preload_libraries = 'safeupdate';
-```
+<Admonition type="warning" className="mt-3">
+  Your project's Postgres version needs to be 15.1.173+ to use this extension. You can check and
+  upgrade your project in the
+  [Settings](https://supabase.com/dashboard/project/tvpxklkoidtevipfmlyv/settings/infrastructure)
```
Comment (soedirgo) [inline apps/docs/content/guides/database/extensions/pg-safeupdate.mdx:16]:
```suggestion
  [Settings](https://supabase.com/dashboard/project/_/settings/infrastructure)
```

**Review by soedirgo — APPROVED** [review]

File: apps/docs/content/guides/database/extensions/pg-safeupdate.mdx [hunk 2]
```diff
@@ -10,19 +10,17 @@ The `pg-safeupdate` extension is a useful tool for protecting data integrity and
...
 
-`pg-safeupdate` can be enabled on a role basis (where `anon` is the role we want to modify in this example):
-
-{/* prettier-ignore */}
-```sql
-GRANT SET ON PARAMETER session_preload_libraries TO anon;
-ALTER ROLE anon SET session_preload_libraries = 'safeupdate';
-```
+<Admonition type="warning" className="mt-3">
+  Your project's Postgres version needs to be 15.1.1.73+ to use this extension. You can check and
```
Comment (charislam) [inline apps/docs/content/guides/database/extensions/pg-safeupdate.mdx:14]: Putting blank lines between the tag and the content will stop Prettier from breaking the lines (which also breaks them in the final HTML):

image

**Review by charislam — APPROVED** [review]
Was that a formatting run causing blog changes? 🤔 If you revert those, it won't be blocked on marketing approval 😄

File: apps/docs/content/guides/database/extensions/pg-safeupdate.mdx [hunk 3]
```diff
@@ -10,19 +10,17 @@ The `pg-safeupdate` extension is a useful tool for protecting data integrity and
...
+  upgrade your project in the
+  [Settings](https://supabase.com/dashboard/project/_/settings/infrastructure)
+</Admonition>
```
Comment (charislam) [inline apps/docs/content/guides/database/extensions/pg-safeupdate.mdx:17]:
```suggestion
<Admonition type="warning" className="mt-3">

Your project's Postgres version needs to be 15.1.1.73+ to use this extension. You can check and upgrade your project in the [Settings](https://supabase.com/dashboard/project/_/settings/infrastructure)

</Admonition>
```

**Review by charislam — APPROVED** [review]

**Comment by joshenlim** [comment]
the page `pg-safeupdate.mdx` no longer exists on the latest master - will hence be closing this PR as the changes here are probably no longer relevant? Please do let me know otherwise 🙏

**Pull request closed without merging**